docs-markdown/src/controllers/cleanup-controller.ts

```diff
@@ -0,0 +1,427 @@
+            }
+        }
+    } catch (error) {
+        postError(error);
+    }
+    return data;
+}
+
+/**
+ * Checks if array has only one item. if so, then return that item.
+ * @param content takes in string data as content and retuns
+ * first item in array if the array only has one item in array.
+ */
+function handleSingleItemArray(content: string | undefined) {
+    if (content && Array.isArray(content) && content.length == 1) {
+        return content[0];
+    }
+}
+
+/**
+ * Takes in data, metadata key, and value to replace,
+ * does a find and replace of the matching metadata tag
+ * @param data string data from parsed file
+ * @param value value to replace regex match of metadata
+ * @param variable metadata key as variable
+ */
+function singleValueMetadata(data: any, variable: string) {
+    let dashRegex = new RegExp(`${variable}:\\s+-\\s(["'\\sA-Za-z0-9\\-\\_]+)$`, 'm');
+    let bracketRegex = new RegExp(`${variable}:\\s+\\[(["'\\sA-Za-z0-9\\-\\_]+)\\]$`, 'm');
+    let dashMatches = dashRegex.exec(data)
+    let bracketMatch = bracketRegex.exec(data)
+    if (dashMatches) {
+        return data.replace(dashRegex, `${variable}: ${dashMatches[1]}`)
+    } else if (bracketMatch) {
+        return data.replace(bracketMatch, `${variable}: ${bracketMatch[1]}`)
+    } else {
+        return data;
+    }
+}
+
+/**
+ * Converts http:// to https:// for all microsoft links.
+ */
+function microsoftLinks() {
+    traverseFiles(workspace.rootPath, (file: string) => {
+        if (file.endsWith(".md")) {
+            try {
+                let data = readFileSync(file, "utf8")
+                data = handleLinksWithRegex(data)
+                writeFileSync(file, data);
+                showStatusMessage("Running Cleanup...");
+            } catch (error) {
+                postError(error);
+            }
+        }
+    })
+}
+
+/**
+ * replaces input data with regex values for microsoft links.
+ * Looks for data that contains microsoft links for docs, azure, msdn, and technet.
+ * Replace the http with https, and remove language specific url.
+ * @param data takes data string as arg
+ */
+function handleLinksWithRegex(data: string) {
+    let docsRegex = new RegExp(/http:\/\/docs.microsoft.com/g)
+    data = data.replace(docsRegex, "https://docs.microsoft.com")
+    let azureRegex = new RegExp(/http:\/\/azure.microsoft.com/g)
+    data = data.replace(azureRegex, "https://azure.microsoft.com")
+    let msdnRegex = new RegExp(/http:\/\/msdn.microsoft.com/g)
+    data = data.replace(msdnRegex, "https://msdn.microsoft.com")
+    let technetRegex = new RegExp(/http:\/\/technet.microsoft.com/g)
+    data = data.replace(technetRegex, "https://technet.microsoft.com")
+    let docsRegexLang = new RegExp(/https:\/\/docs.microsoft.com\/[A-Za-z]{2}-[A-Za-z]{2}\//g)
+    data = data.replace(docsRegexLang, "https://docs.microsoft.com/")
+    let azureRegexLang = new RegExp(/https:\/\/azure.microsoft.com\/[A-Za-z]{2}-[A-Za-z]{2}\//g)
```

Comment:
The one time i've had to "undo" this is when it was embedded in sample JSON response from the translation API. Probably not worth coding around that though